chaoshub/cli.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+import io
+import json
+from logzero import logger
+
+import click
+
+__all__ = ["login"]
+
+from chaoshub.publish import publish_to_hub
+from chaoslib.settings import load_settings, save_settings
+from chaoslib.types import Settings
+
+
+@click.command()
+@click.argument('login_target')
+@click.pass_context
+def login(ctx: click.Context, login_target: str):
+    """
+    Login to a Chaos Hub.
+    """
+    settings = load_settings()
+    hub_url = click.prompt(click.style("Chaos Hub Url", fg='green'), type=str)
+    token = click.prompt(
+        click.style("Chaos Hub Token", fg='green'), type=str, hide_input=True)
+    update_chaos_hub_settings_in_place(hub_url, token, settings)
+    save_settings(settings)
+    click.echo("Chaos Hub details saved")
+
+
+@click.command()
+@click.argument('journal')
+@click.option('--organisation', default="default",
+              help='Organisation to push the experiment results to.')
+@click.option('--workspace', default="default",
+              help='Workspace to push the experiment results to.')
+@click.pass_context
+def publish(ctx: click.Context, journal: str, organisation: str,
+            workspace: str):
+    """
+    Publish your experiment's findings to a Chaos Hub.
+
+    \b
+    In order to benefit from these features, you must have registered with
+    a Chaos Hub and retrieved an access token. You should set that token in the
+    configuration file with `chaos login hub`.
+    """
+    settings = load_settings()
+    hub_url = settings.get('vendor', {}).get('chaoshub', {}).get('hub_url')
+    token = settings.get('vendor', {}).get('chaoshub', {}).get('token')
+    if hub_url and token:
+        with io.open(journal) as f:
+            journal_json = json.load(f)
+            publish_to_hub(hub_url, token, organisation, workspace, journal,
+                           journal_json)
+    else:
+        logger.warning("No Chaos Hub configured. Please execute " +
+                       "`chaos login hub` before attempting to publish.")
+
+
+def update_chaos_hub_settings_in_place(hub_url: str,
```

Comment:
I try (now anyway ;)) to pslit the CLI definition (@click...) from the functions doing the job. In different Python modules. wdyt?

Reply:
If you mean that the `update_chaos_hub_settings_in_place` should go elsewhere, in its own module, then that makes sense. Is that what you mean?